irispy/iris_tools.py

```diff
@@ -253,25 +258,25 @@ def get_iris_response(pre_launch=False, response_file=None, response_version=Non
             for k in range(n_time_obs):
                 iris_response["AREA_SG"][1,w_nuv] = scipy.interpolate.interp1d(iris_fit_nuv, iris_response["LAMBDA"][:, k], kind='cubic')
 
-    # 3. SJI effective areas
-    if int(iris_response["VERSION"]) = 3:
+        # 3. SJI effective areas
         shp_sji = iris_response["COEFFS_SJI"].shape
         area_pre_launch = iris_response["GEOM_AREA"]
         for j in range(shp_sji[0]):
             # Calculate pre-launch area from the individual elements
             for k in range(len(iris_response["INDEX_EL_SJI"][j, :])):
-                area_pre_launch = iris_response["GEOM_AREA"] * np.prod(iris_response["INDEX_EL_SJI"][j, k]["trans"], axis=correct_axis)
+                #area_pre_launch = iris_response["GEOM_AREA"] * np.prod(iris_response["INDEX_EL_SJI"][j, k]["trans"], axis=correct_axis)
+                area_pre_launch = iris_response["GEOM_AREA"] * np.prod(iris_response["ELEMENTS"]["trans"][iris_response["INDEX_EL_SJI"]], axis=1)[j, k]
```

Comment:
This should be moved to and replace the original definition [here](https://github.com/sunpy/irispy/pull/108/files/fe1f1daf7d3d4f7d8333a2c9961960d7300a5f64..e43a794df5b3b9b2a30298a2f6f989a6fa3437f7#diff-dee23d5475a6ed5e368938455c92fb2eR263)